src/Units/UnitContext.java

```diff
@@ -125,12 +125,8 @@ public void setWeapon(WeaponModel w)
     {
       rangeMin = w.rangeMin;
       rangeMax = w.rangeMax;
-      // Build a scratch UnitContext because we don't trust UnitMods with our state
-      UnitContext uc = new UnitContext(this);
       for( UnitModifier mod : mods )
-        mod.modifyAttackRange(uc);
-      rangeMin = uc.rangeMin;
-      rangeMax = uc.rangeMax;
+        mod.modifyAttackRange(this);
```

Comment:
Is there ever a case where we might call `setWeapon` more than once on the same `UnitContext`? Should we explicitly make that an error case?

Reply:
I don't think so, but it's not really an error case.
We set the range to our new weapon's range before modification, and the modifiers shouldn't be changing anything but min/max range.